src/main/java/com/walking/intensive/chapter2/task8/Task8.java

```diff
@@ -24,7 +24,35 @@ public static void main(String[] args) {
     }
 
     static double getHappyTicketChance() {
-        // Ваш код
-        return 0.0;
+        int leftDigits = 0;
+        int rightDigits = 0;
+        double getEquality = 0;
+        double getProbability;
+
+        for (int i = 0; i < 1000; i++) {
+            if (i < 10) {
+                leftDigits = (i % 10);
+            } else if (i < 99) {
+                leftDigits = i / 10 + (i % 10);
+            } else {
+                leftDigits = i / 100 + (i % 100) / 10 + (i % 10);
+            }
+            for (int j = 1; j < 1000; j++) {
+                if (leftDigits == rightDigits) {
+                    getEquality++;
+                }
+                {
+                    if (j < 10) {
+                        rightDigits = (j % 10);
+                    } else if (j < 99) {
+                        rightDigits = j / 10 + (j % 10);
+                    } else {
+                        rightDigits = j / 100 + (j % 100) / 10 + (j % 10);
+                    }
+                }
+            }
+        }
+
+        return getProbability = getEquality / (999999 + 1);
```

Comment:
Error: eckstyle] [ERROR] /home/runner/work/intensive-tasks-2024/intensive-tasks-2024/src/main/java/com/walking/intensive/chapter2/task8/Task8.java:56:31: Inner assignments should be avoided. [InnerAssignment]